### DISCUSSION TITLE
Adding prerequisites for the course

### DISCUSSION THREAD

**Discussion by snehal0904** [Sunbird-Ed/Community#262] [Enhancement request]
Dear Sunbird Team,
Proposing a feature to add prerequisites for the course content in sunbird ED.
We are using sunbird ED for multiple use cases. In one of our use-case, we want students to complete the prerequisites before they jump into the next content. For example, consider we have a mathematics course where we have units like addition, subtraction, division, square root, etc. Here we want students to complete subtraction and division before they jump to the square root unit. In this case, we can add completion prerequisites
 
Prerequisites can be of multiple types, below are the possible 
 
- Content gets unlocked when pre-requisite content is viewed
- Content gets unlocked when a pre-requisite assessment is passed (with min. score requirement)
- Content gets unlocked x days after enrolling in the course
- Content gets unlocked x days after completing a pre-requisite content
- Content gets unlocked on a specific date

**Approach**

- Add a prerequisites JSON type field against the course. 

   In this JSON add the prerequisites conditions which are required to access the content of the course
   Example JSON for prerequisites can be -
```
{
"prerequisite": [
  “do_1312435345”: {
    “condition”: “complete_content”,
    “dependantContentIds”: [
      “do_1312435334”,
      “do_2322456676”
       ]
  },
  “do_1312435346”: {
    “condition”: “assessment_pass”,
    “dependantContentIds”: [
      “do_1312435334”
    ]
  }
]
}
```

- On the course detail page get prerequisites and apply them to the lesson display. If prerequisite conditions are not completed content will display locked.

- On the course playlist, consider prerequisites to show the next content

Please share your feedback or any other inputs to achieve this functionality.

 @Krishnaj20 @rhwarrier @coolbung @parthlawate

**Comment by rhwarrier** [comment]
pls refer to https://github.com/orgs/Sunbird-Ed/discussions/200#discussioncomment-4216007 where we'll attempt to consolidate needs raised around Courses and take the design forward.
